### DISCUSSION TITLE
lock-after-session-end

### DISCUSSION THREAD

**Discussion by mobilete** [rustdesk/rustdesk#9448] [General]
### Bug Description

having the option lock-after-session-end enabled on my client, when I connect to a remote machine and press the X to close, the remote machine does not lock itself.

i am using rustdesk version 1.3.0 as client iOs version 17.6.1
remote machine is rustdesk 1.2.6 and win 10 22h2

### How to Reproduce

tap rustdesk app icon
tap the host to establish connection
swap app to get my 2fa code and enter my 2fa in rustdesk prompt
tap the x on the bottom left, confirm ok when asked "are you sure you want to close the connection?"

### Expected Behavior

Expect to have the remote machine to lock itself like equivalernt of "Windows Key + L"

### Operating system(s) on local side and remote side

iOs 17.6.1 --> Windows10.22h2

### RustDesk Version(s) on local side and remote side

1.3.0 --> 1.2.6

### Screenshots

![image](https://github.com/user-attachments/assets/89f5dbd9-507a-4063-8c2d-9207c124f730)

### Additional Context

refering to this feature
https://rustdesk.com/docs/en/self-host/client-configuration/advanced-settings/#lock-after-session-end

**Comment by rustdesk** [comment]
`lock-after-session-end` is an option of controlling side.

**Comment by mobilete** [comment]
To clarify if I understand correctly, do you mean that this option is locking the controlling side session ?

**Comment by rustdesk** [comment]
I meant your controlling side is iOS, we have no custom client for iOS, how do you use this `lock-after-session-end`?

**Comment by rustdesk** [comment]
> To clarify if I understand correctly, do you mean that this option is locking the controlling side session ?

It is used on controlling side, but it locks controlled side.

**Comment by mobilete** [comment]
I thought that enabling this option on the controller side would lock the remote session when you disconnect.

At least that's what I remember when using Windows to  Windows but i just checked another couple of machine and the remote session is not locked and my local windows is also not locking.

I am not too sure to understand this feature and how it is used.

Thanks

**Comment by rustdesk** [comment]
Make sure the following setting is turned on in your iOS device. The default option (as shown in the screenshot you provided) only works for new connections. When you first connect, it will use the overridden default option, and it will continue to use it for subsequent connections. However, for existing connections, it does not apply.

![image](https://github.com/user-attachments/assets/4c9be43a-ee67-4449-aab4-6d339fa809d3)

**Comment by kaka77** [comment]
+1 , I've encountered this issue too and haven't figured out how to resolve it yet. Have you guys solved it?

**Comment by karyeet** [comment]
I am encountering the opposite issue. 
Despite having the option unchecked on client and host, the device always locks after ending the session.

**Comment by Probrogit** [comment]
Can confirm Windows 11 25H2 locks after disconnect with any lock-after-session-end settings. Checked or not.

**Comment by Nickperillo** [comment]
I have the same issue but with macOS. Checking "Lock after session end" does nothing after you close the session.
